ps_healthcheck.php

```diff
@@ -0,0 +1,186 @@
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors.
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+if (!defined('_CAN_LOAD_FILES_')) {
+    exit(1);
+}
+
+if (file_exists(__DIR__ . '/vendor/autoload.php')) {
+    require_once __DIR__ . '/vendor/autoload.php';
+}
+
+use PrestaShop\Module\HealthCheck\Repository\HealthCheckConfigRepository;
+use PrestaShop\PrestaShop\Adapter\SymfonyContainer;
+
+/**
+ * Class Ps_Healthcheck.
+ */
+class Ps_Healthcheck extends Module
+{
+    const MODULE_NAME = 'ps_healthcheck';
+    /**
+     * @var HealthCheckConfigRepository
+     */
+    private $repository;
+
+    public function __construct()
+    {
+        $this->name = static::MODULE_NAME;
+        $this->author = 'PrestaShop';
+        $this->version = '0.1.0';
+        $this->need_instance = 0;
+        $this->tab = 'settings';
+        $this->tabs = [
+            [
+                'class_name' => 'AdminHealthCheck',
+                'visible' => true,
+                'name' => 'Health Check',
+                'parent_class_name' => 'AdminParentPreferences',
+            ],
+        ];
+
+        $this->bootstrap = true;
+        parent::__construct();
+
+        $this->description = $this->trans('Adds an endpoint to perform application health check.', [], 'Modules.Healthcheck.Admin');
+        $this->secure_key = Tools::encrypt($this->name);
+
+        $this->ps_versions_compliancy = ['min' => '1.7.8.0', 'max' => _PS_VERSION_];
+    }
+
+    public function install()
+    {
+        if (!parent::install()) {
+            return false;
+        }
+
+        $installed = false;
+        if (null !== $this->getRepository()) {
+            $installed = $this->installFixtures();
+        }
+
+        if ($installed && $this->installTab()) {
+            return true;
+        }
+
+        $this->uninstall();
+
+        return false;
+    }
+
+    public function enable($force_all = false)
+    {
+        if (!$this->installTab()) {
+            return false;
+        }
+
+        return parent::enable($force_all);
+    }
+
+    /**
+     * @return bool
+     *
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    private function installFixtures()
+    {
+        $installed = true;
+        $errors = $this->getRepository()->createTables();
+        if (!empty($errors)) {
+            $this->addModuleErrors($errors);
+            $installed = false;
+        }
+
+        $errors = $this->getRepository()->installFixtures();
+        if (!empty($errors)) {
+            $this->addModuleErrors($errors);
+            $installed = false;
+        }
+
+        return $installed;
+    }
+
+    /**
```

Comment:
Did we really need of this method because we are compatible 1.7.8+ ?

Reply:
you are right, I dont need a specific entry in the left menu ;) thanks